pymc/tests/test_distributions_timeseries.py

```diff
@@ -12,20 +12,118 @@
 import pytest
 
+from scipy import stats
+
+import pymc as pm
+
 from pymc.aesaraf import floatX
 from pymc.distributions.continuous import Flat, Normal
-from pymc.distributions.timeseries import AR, AR1, GARCH11, EulerMaruyama
+from pymc.distributions.timeseries import (
+    AR,
+    AR1,
+    GARCH11,
+    EulerMaruyama,
+    GaussianRandomWalk,
+    gaussianrandomwalk,
+)
 from pymc.model import Model
 from pymc.sampling import sample, sample_posterior_predictive
 from pymc.tests.helpers import select_by_precision
 
-# pytestmark = pytest.mark.usefixtures("seeded_test")
-pytestmark = pytest.mark.xfail(reason="Timeseries not refactored")
+
+class TestGaussianRandomWalk:
+    @pytest.mark.parametrize(
+        "kwargs,expected",
+        [
+            ({"steps": 5}, (6,)),
+            ({"size": 1}, (5,)),
+            ({"size": 2}, (5,)),
+            # implied dims are not working
+            pytest.param({"mu": [0, 0]}, (2, 5), marks=pytest.mark.xfail),
+        ],
+    )
+    def test_grw_rv_op_shape(self, kwargs, expected):
+        """Basic test for GRW RV op"""
+        default_kwargs = dict(init=1, mu=3, sd=0.0000001, steps=4, size=None)
+
+        combined_kwargs = {**default_kwargs, **kwargs}
+        grw = gaussianrandomwalk(
+            combined_kwargs["mu"],
+            combined_kwargs["sd"],
+            combined_kwargs["init"],
+            combined_kwargs["steps"],
+        ).eval()
+
+        assert grw.shape == expected
+
+    def test_grw_logp(self):
+        # `at.diff` is currently broken with constants
+        test_vals = [0, 1, 2]
+        vals = at.vector("vals")
+        mu = 1
+        sigma = 1
+        init = pm.Normal.dist(0, sigma)
+
+        with pm.Model():
+            grw = GaussianRandomWalk("grw", mu, sigma, init=init, steps=2)
+
+        logp = pm.logp(grw, vals)
+        logp_eval = logp.eval({vals: test_vals})
+
+        logp_reference = (
+            stats.norm(0, sigma).logpdf(test_vals[0])
+            + stats.norm(mu, sigma).logpdf(np.diff(test_vals)).sum()
+        )
+
+        np.testing.assert_almost_equal(logp_eval, logp_reference)
+
+    def test_grw_inference(self):
+        mu, sigma, steps = 2, 1, 10000
+        obs = np.concatenate([[0], np.random.normal(mu, sigma, size=steps)]).cumsum()
+
+        with pm.Model():
+            _mu = pm.Uniform("mu", -10, 10)
+            _sigma = pm.Uniform("sigma", 0, 10)
+            # Workaround for bug in `at.diff` when data is constant
+            obs_data = pm.MutableData("obs_data", obs)
+            grw = GaussianRandomWalk("grw", _mu, _sigma, steps=steps, observed=obs_data)
+
+            trace = pm.sample()
+
+        recovered_mu = trace.posterior["mu"].mean()
+        recovered_sigma = trace.posterior["sigma"].mean()
+        np.testing.assert_allclose([mu, sigma], [recovered_mu, recovered_sigma], atol=0.2)
+
+    @pytest.mark.parametrize(
+        "steps,size,expected",
+        (
+            (1, None, (2,)),
+            (2, 1, (1, 3)),
```

Comment:
This fails now. Obviously because we added shapes, but does this mean now that  shape from params function needs to be updated to account for size

https://github.com/pymc-devs/pymc/pull/5298/files#diff-5fece42aa9647732b360b6c0e13de089a8408c68ae38b650fba38a1c40322229R52

<img width="748" alt="image" src="https://user-images.githubusercontent.com/7213793/156865913-a27eecd8-3702-4865-82f8-8b66cdd5bf7f.png">

Reply:
Have you updated Aesara? The meaning of size for multivariate dists changed and should now mean this test is correct.

Reply:
That  was the issue thanks